src/System.Private.CoreLib/shared/System/Runtime/Intrinsics/Vector128.cs

```diff
@@ -40,5 +40,29 @@ private unsafe string DisplayString
                 }
             }
         }
+
+        public Vector128<byte> AsByte() => AsByte();
```

Comment:
I think all the intrinsic in `VectorXXX` classes should have software fallback.

Reply:
I agree - to make this work for well for Mono and ProjectN/CoreRT.

Reply:
What is the point of adding software fallbacks for these methods? They are not actually usable for anything without corresponding intrinsic support provided by the given runtime.

Reply:
I think these types/methods are usable even without the intrinsic support provided by the given runtime.

If somebody tries to use these methods and the runtime does not have the support, it will end up in infinite loop or stackoverflow with the current implementation. I do not think it is the desired behavior.

The software fallbacks are trivial to implement for these. Change `public Vector128<byte> AsByte() => AsByte();` to `public Vector128<byte> AsByte() => Unsafe.As<Vector<T>, Vector<byte>>(ref this);` and so on.

Reply:
> I think these types/methods are usable even without the intrinsic support provided by the given runtime.

Not really. The types are currently opaque, and even with the few helper methods being moved here, you will only be able to bit-blit between different `T` and potentially be able to (inefficiently) create an instance with the correct value.
You still won't be able to actually access the individual elements or do anything else other than inefficiently pass the data around.

> If somebody tries to use these methods and the runtime does not have the support, it will end up in infinite loop or stackoverflow with the current implementation. I do not think it is the desired behavior.

The same is true for any of the other intrinsics under the `System.Runtime.Intrinsics` namespace. We can either split this into `VectorXXX.cs` and `VectorXXX.PlatformNotSupported.cs`, and tell the other runtimes to only compile the `*.PlatformNotSupported.cs` files (until they add support) or we could just exclusively have these throw `PNSE` and rely on the runtime intrinsic recognition to cause expansion (we would also need to find some way to continue supporting indirect invocation, such as via reflection).

> The software fallbacks are trivial to implement for these

Agreed. However, the explicit design decision so far is (and has consistently been) that we would not be providing software fallbacks for any intrinsics or intrinsic helpers.
Instead, users would be expected to provide their own types for the public surface area and would use the `Load/Store` intrinsics to interop between the two (because they have to do this anyways to work on unsupported platforms).

Reply:
> However, the explicit design decision so far is (and has consistently been) that we would not be providing software fallbacks for any intrinsics or intrinsic helpers.

This design decision makes sense for the actual hardware intrinsics that map to instruction - to make sure that the software fallback does not deviate from the actual hardware instruction behavior.

It is the exact opposite for platform neutral intrinsics. The platform neutral intrinsics do have software fallbacks when possible (there are only a few where the software fallback is not possible). These types are platform neutral, so I think they can be on the platform neutral intrinsic plan just fine.

Reply:
Fixed.